core/commands/storage.go

```diff
@@ -179,6 +171,11 @@ Receive proofs as collateral evidence after selected nodes agree to store the fi
 		var peers []string
 		// init retry queue
 		retryQueue := storage.NewRetryQueue(int64(len(peers)))
+		// set price limit, the price is default when host doesn't provide price
+		price, found := req.Options[uploadPriceOptionName].(int64)
+		if found && price <= 0 {
+			return fmt.Errorf("cannot input a negative price")
```

Comment:
"price limit should higher than zero"?

Reply:
if you setting zero price, and in custom can't use no hub price, are you expecting pay nothing to host?? Then I think no host is willing to sign on the contract

Reply:
I mean, should update the error msg to something like ("price limit should higher than ", HostPriceLowBoundary)